src/common/http/client.go

```diff
@@ -28,41 +28,80 @@ import (
 	"reflect"
 )
 
-// Client is a util for common HTTP operations, such Get, Head, Post, Put and Delete.
-// Use Do instead if  those methods can not meet your requirement
-type Client struct {
-	modifiers []modifier.Modifier
-	client    *http.Client
-}
+const (
+	// DefaultTransport used to get the default http Transport
+	DefaultTransport = iota
+	// InsecureTransport used to get the insecure http Transport
+	InsecureTransport
+	// InternalTransport used to get the internal secure http Transport
+	InternalTransport
```

Comment:
Would it be simpler and more consistent if we merge `internal` and `secure` transport into one `secure transport`?

This `secure transport` should trust the CA in the truststore OS as well as the Harbor internal CA if it's configured?